### PR TITLE
1201 make risc0 public input optional

### LINKED ISSUE #1201: fix(cli): Aligned CLI should accept Public Inputs as an optional input for Risc0
The Aligned submit CLI currently treats the public input argument for Risc0 as a required argument even though it is optional. We should submitting Risc0 proofs with no public input.

ref:
https://github.com/yetanotherco/aligned_layer/blob/testnet/batcher/aligned/src/main.rs#L493
https://github.com/yetanotherco/aligned_layer/blob/testnet/batcher/aligned/src/main.rs#L574

### PR DESCRIPTION
**Pull request by PatStiles**
closes #1201 

## To Test:
- Run local testnet
- Submit risc0 proofs to the local batcher via `make batcher_send_risc0_burst`. Observe they are processed.
- Submit risc0 proof with empty public inputs to the local batcher via `make batcher_send_risc0_task_no_pub_input`. Observe they are processed.

### CODE REVIEW

**Review by uri-99 — APPROVED** [review]

**Review by MarcosNicolau — APPROVED** [review]
Tested on macos and linux.

File: batcher/aligned-batcher/src/zk_utils/mod.rs [hunk 1]
```diff
@@ -65,21 +65,25 @@ fn verify_internal(verification_data: &VerificationData) -> bool {
...
-                let mut image_id = [0u8; 32];
-                image_id.copy_from_slice(image_id_slice.as_slice());
-                return verify_risc_zero_proof(
-                    verification_data.proof.as_slice(),
-                    &image_id,
-                    pub_input,
-                );
+            // Allow for empty public input
+            let Some(image_id_slice) = &verification_data.vm_program_code else {
+                warn!("Trying to verify Risc0 proof but image id. Returning false");
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/zk_utils/mod.rs]:
```suggestion
                warn!("Trying to verify Risc0 proof but image id was not provided. Returning false");
```

File: batcher/aligned-batcher/src/zk_utils/mod.rs [hunk 2]
```diff
@@ -65,21 +65,25 @@ fn verify_internal(verification_data: &VerificationData) -> bool {
...
-                &verification_data.pub_input,
-            ) {
-                let mut image_id = [0u8; 32];
-                image_id.copy_from_slice(image_id_slice.as_slice());
-                return verify_risc_zero_proof(
-                    verification_data.proof.as_slice(),
-                    &image_id,
-                    pub_input,
-                );
+            // Allow for empty public input
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/zk_utils/mod.rs]: This comment doesn't add much value here

File: batcher/aligned-batcher/src/zk_utils/mod.rs [hunk 3]
```diff
@@ -65,21 +65,25 @@ fn verify_internal(verification_data: &VerificationData) -> bool {
...
+            let Some(image_id_slice) = &verification_data.vm_program_code else {
+                warn!("Trying to verify Risc0 proof but image id. Returning false");
+                return false;
+            };
+
+            // Risc0 can have 0 public input. In which case we supply an empty Vec<u8>.
+            let mut public_input = &vec![];
+            if let Some(pub_input) = &verification_data.pub_input {
+                public_input = pub_input;
             }
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/zk_utils/mod.rs]:
```suggestion
            let pub_input = verification_data.pub_input.clone().unwrap_or_default();
```
and then the reference is put to `pub_input` when calling `verify_risc_zero_proof`

**Comment by entropidelic** [comment]
maybe a `generate_risc_zero_empty_journal_proof` make target will be useful like we have for other proofs

**Comment by entropidelic** [comment]
after running `make deps` and `make install_aligned_compiling` everything worked fine. Left some small nits on the code

**Review by Oppen — APPROVED** [review]
Seems to work. Other than addressing Marian's comments, it looks good.

File: batcher/aligned-batcher/src/zk_utils/mod.rs (hunk 3 above)
Comment (PatStiles) [reply batcher/aligned-batcher/src/zk_utils/mod.rs]: Addressed!

File: batcher/aligned-batcher/src/zk_utils/mod.rs (hunk 2 above)
Comment (PatStiles) [reply batcher/aligned-batcher/src/zk_utils/mod.rs]: Addressed

File: batcher/aligned-batcher/src/zk_utils/mod.rs (hunk 1 above)
Comment (PatStiles) [reply batcher/aligned-batcher/src/zk_utils/mod.rs]: Addressed!

File: Makefile [hunk 4]
```diff
@@ -508,6 +518,11 @@ generate_sp1_fibonacci_proof:
 	@mv scripts/test_files/sp1/fibonacci_proof_generator/script/sp1_fibonacci.proof scripts/test_files/sp1/
 	@echo "Fibonacci proof and ELF generated in scripts/test_files/sp1 folder"
 
+generate_risc_zero_empty_journal_proof:
+	@cd scripts/test_files/risc_zero/no_public_inputs && RUST_LOG=info cargo run --release
+	@echo "Fibonacci proof and ELF generated with empty journal generated in scripts/test_files/risc_zero/no_public_inputs folder"
```
Comment (entropidelic) [inline Makefile]:
```suggestion
	@echo "Fibonacci proof and ELF with empty journal generated in scripts/test_files/risc_zero/no_public_inputs folder"
```

**Pull request merged**